iroh-bytes/src/store/file/test_support.rs

```diff
@@ -0,0 +1,399 @@
+//! DB functions to support testing
+//!
+//! For some tests we need to modify the state of the store in ways that are not
+//! possible through the public API. This module provides functions to do that.
+use std::{
+    io,
+    path::{Path, PathBuf},
+};
+
+use futures::channel::oneshot;
+
+use super::{
+    tables::{ReadableTables, Tables},
+    ActorError, ActorMessage, ActorResult, ActorState, DataLocation, EntryState, FilterPredicate,
+    OutboardLocation, OuterResult, Store, StoreInner,
+};
+use crate::{
+    store::{
+        bao_file::{raw_outboard_size, SizeInfo},
+        DbIter,
+    },
+    Hash,
+};
+use redb::ReadableTable;
+
+/// The full state of an entry, including the data.
+#[derive(derive_more::Debug)]
+pub enum EntryData {
+    /// Complete
+    Complete {
+        /// Data
+        #[debug("data")]
+        data: Vec<u8>,
+        /// Outboard
+        #[debug("outboard")]
+        outboard: Vec<u8>,
+    },
+    /// Partial
+    Partial {
+        /// Data
+        #[debug("data")]
+        data: Vec<u8>,
+        /// Outboard
+        #[debug("outboard")]
+        outboard: Vec<u8>,
+        /// Sizes
+        #[debug("sizes")]
+        sizes: Vec<u8>,
+    },
+}
+
+impl Store {
+    /// Get the complete state of an entry, both in memory and in redb.
+    #[cfg(test)]
```

Comment:
why not make the whole module `cfg[test]`?

Reply:
It is used from the cli tests, which are not in the same crate.